Bluepill-cli/Bluepill-cli/Simulator/BPSimulator.m

```diff
@@ -338,47 +341,39 @@ - (void)deleteSimulatorWithCompletion:(void (^)(NSError *error, BOOL success))co
+        completion(nil, NO);
+        return;
+    }
     if (self.app) {
         [BPUtils printInfo:INFO withString:@"Terminating Simulator.app"];
         [self.app terminate];
-        // We need to wait until the simulator has shut down.
-        int attempts = 300;
-        while (attempts > 0 && ![self.device.stateString isEqualToString:@"Shutdown"]) {
-            [NSThread sleepForTimeInterval:1.0];
-            --attempts;
-        }
-        if (![self.device.stateString isEqualToString:@"Shutdown"]) {
-            [BPUtils printInfo:ERROR withString:@"Timed out waiting for %@ to shutdown. It won't be deleted. Last state: %@", self.device.name, self.device.stateString];
-            // Go ahead and try to delete anyway
-        }
-        [deviceSet deleteDeviceAsync:self.device completionHandler:^(NSError *error) {
-            if (error) {
-                [BPUtils printInfo:ERROR withString:@"Could not delete simulator: %@", [error localizedDescription]];
-            }
-            completion(error, error ? NO: YES);
-        }];
     } else if (self.device) {
-        [BPUtils printInfo:INFO withString:@"Shutting down Simulator Device"];
-        [self.device shutdownAsyncWithCompletionHandler:^(NSError *error) {
-            if (!error) {
-                [BPUtils printInfo:INFO withString:@"Deleting Simulator Device"];
-                dispatch_async(dispatch_get_main_queue(), ^{
-                    [deviceSet deleteDeviceAsync:self.device completionHandler:^(NSError *error) {
-                        if (error) {
-                            [BPUtils printInfo:ERROR withString:@"Could not delete simulator: %@", [error localizedDescription]];
-                        }
-                        completion(error, error ? NO: YES);
-                    }];
-                });
-            } else {
-                [BPUtils printInfo:ERROR withString:@"Could not shutdown simulator: %@", [error localizedDescription]];
-                completion(error, NO);
-            }
-        }];
-    } else {
-        [BPUtils printInfo:ERROR withString:@"No device to delete"];
-        completion(nil, NO);
+        [BPUtils printInfo:INFO withString:@"Shutting down Simulator"];
+        [self.device shutdownWithError:&error];
+        if (error) {
+            [BPUtils printInfo:ERROR withString:@"Shutting down Simulator failed: %@", [error localizedDescription]];
+            completion(error, NO);
+            return;
+        }
+    }
+    // We need to wait until the simulator has shut down.
+    int attempts = 300;
+    while (attempts > 0 && ![self.device.stateString isEqualToString:@"Shutdown"]) {
+        [NSThread sleepForTimeInterval:1.0];
+        --attempts;
+    }
+    if (![self.device.stateString isEqualToString:@"Shutdown"]) {
+        [BPUtils printInfo:ERROR withString:@"Timed out waiting for %@ to shutdown. It won't be deleted. Last state: %@", self.device.name, self.device.stateString];
```

Comment:
I know this was existing code, but could we change the message to state that `It may not be possible to delete simulator %@`?